### PR TITLE
Clean up cachedActivity after destroyed it.

### PR DESCRIPTION
**Pull request by PaladiRka**
https://github.com/KasperskyLab/Kaspresso/issues/387

### CODE REVIEW

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/languages/LanguageImpl.kt [hunk 1]
```diff
@@ -23,6 +23,9 @@ class LanguageImpl(
             if (stage == Stage.CREATED) {
                 cachedActivity = activity
             }
+            if ((stage == Stage.DESTROYED) and (cachedActivity == activity)) {
```
Comment (RuslanMingaliev) [inline kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/languages/LanguageImpl.kt:26]: AFAIK `onDestroy` method call is not guaranteed. What about  `Stage.DESTROYED`?  
May be weak reference on `cachedActivity` is more suitable here?

**Review by RuslanMingaliev — DISMISSED** [review]

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/languages/LanguageImpl.kt (hunk 1 above)
Comment (PaladiRka) [reply kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/languages/LanguageImpl.kt]: Yes, is better

**Comment by PaladiRka** [comment]
@RuslanMingaliev

**Review by RuslanMingaliev — APPROVED** [review]

**Review by AzamatCherchesov — APPROVED** [review]

**Comment by e13mort** [comment]
@matzuk @eakurnikov could you review this MR please?

**Review by eakurnikov — APPROVED** [review]

**Comment by PaladiRka** [comment]
Can you tell when this fix will released?

**Comment by AzamatCherchesov** [comment]
@PaladiRka this fix will be included in release 1.4.2, which we plan to publish in a few days

**Pull request merged**